src/slicer.h

```diff
@@ -518,9 +528,23 @@ class Slicer
      * \param z The Z coordinate of the layer to intersect with.
      * \return A slicer segment.
      */
-    SlicerSegment project2D(const Point3& p0, const Point3& p1, const Point3& p2, const coord_t z) const;
+    static SlicerSegment project2D(const Point3& p0, const Point3& p1, const Point3& p2, const coord_t z);;
+
+    /*! Create an array of "z bboxes" for each face. */
+    static std::vector<std::pair<int32_t, int32_t>> buildZHeightsForFaces(const Mesh &mesh);
+
+    /*! Create the polygons in layers. */
+    static void makePolygons(Mesh &mesh, SlicingTolerance slicing_tolerance, std::vector<SlicerLayer>& layers);
+
+    /* Creates a vector of layers and set their z value. */
+    static std::vector<SlicerLayer> buildLayersWithHeight(size_t slice_layer_count, SlicingTolerance slicing_tolerance,
+        coord_t initial_layer_thickness, coord_t thickness, bool use_variable_layer_heights,
+        const std::vector<AdaptiveLayer> *adaptive_layers);
+
+    /* Creates the segments and write them into the layers. */
```

Comment:
Could you write these documentation entries in Doxygen's style?

Reply:
fixed